src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -378,8 +378,8 @@ protected function mergeWithExisting($entitiesByModule)
             // Merge
             if ($existingMessages) {
                 $entitiesByModule[$module] = array_merge(
-                    $existingMessages,
-                    $messages
+                    $messages,
```

Comment:
Sorry, looking at the context of the PR title, I think we might have different understandings of this behaviour. The intent of merge is to not delete any keys not present in the new strings (in case older versions still need to localise those values), but not to prevent changed ones from being updated.

If you do merge=0, keys that aren't found are deleted, meaning any subsequent push to transifex deletes both the master, and all localisations for that key. It's normally important to keep those keys if we release, for instance, a new 3.4.x tag, even though a new 3.6.x tag might have removed that key.